input/otlp/logs.go

```diff
@@ -109,15 +110,54 @@ func (c *Consumer) convertLogRecord(
 		}
 		event.Span.ID = spanID.HexString()
 	}
-	if attrs := record.Attributes(); attrs.Len() > 0 {
-		setLabels(attrs, &event)
+	attrs := record.Attributes()
+	setLabels(attrs, &event)
+	if event.Error != nil {
+		event.Processor = model.ErrorProcessor
+		event.Event.Kind = "event"
+		event.Event.Type = "error"
 	}
 	return event
 }
 
 func setLabels(m pcommon.Map, event *model.APMEvent) {
```

Comment:
I don't think we should change `setLabels`. Its job is meant to be set to set `labels.*` key/value pairs, and nothing else. Also, changing it means that the `setLabels(body.Map(), &event)` call on line 100/101 above would change behaviour.

Instead of doing this, how about we iterate over the attributes in this method, and just call `setLabel` for the attributes that aren't interpreted? That's what we do for spans.

Reply:
Agree with the first part.

> Instead of doing this, how about we iterate over the attributes in this method, and just call setLabel for the attributes that aren't interpreted? That's what we do for spans.

Does "this method" refer to `setLabels` ? 

___

I've pushed some changes

Reply:
Sorry for being unclear, I meant in `convertLogRecord`.

Reply:
No worries, I've pushed some changes :+1: